src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -347,8 +353,18 @@ public void snapshotCreateBQDataset(BigQueryProject bigQueryProject, Snapshot sn
         if (bigQueryProject.datasetExists(snapshotName)) {
             bigQueryProject.deleteDataset(snapshotName);
         }
+
+        // TODO: When we support multiple datasets per snapshot, this will need to be reworked
+        String representativeRegion = snapshot.getSnapshotSources()
+            .get(0)
+            .getDataset()
+            .getDatasetSummary()
+            .getStorage()
+            .stream()
+            .collect(Collectors.toMap(StorageResource::getCloudResource, StorageResource::getRegion))
+            .get(GoogleCloudResource.BIGQUERY.toString());
```

Comment:
Using collect(Map).get() seems like an odd way to find a specific value. Could you use filter().findFirst() instead?